Sources/ApolloCodegenLib/IR/IR+EntitySelectionTree.swift

```diff
@@ -443,7 +443,29 @@ extension IR.EntitySelectionTree.EnclosingEntityNode {
     withNodeRootType nodeRootType: GraphQLCompositeType,
     using entityStorage: IR.RootFieldEntityStorage
   ) {
-    switch otherTree.rootNode.child {
+    let rootNode = otherTree.rootNode
```

Comment:
This fixes a bug where, when merging the entity selection tree from a fragment, we forgot to merge type conditions that were at the root of the fragment.